components/brave_referrals/browser/brave_referrals_service_unittest.cc

```diff
@@ -0,0 +1,192 @@
+            response_body =
+                "{\"download_id\":\"" + std::string(kTestDownloadId) + "\"}";
+          } else if (request.url.path_piece() == "/promo/activity") {
+            // Respond with a successful finalization check response
+            response_body = "{\"finalized\":true}";
+          }
+
+          url_loader_factory_.AddResponse(request.url.spec(), response_body);
+          request_count_++;
+        }));
+
+    BraveReferralsService::SetReferralInitializedCallbackForTesting(
+        new BraveReferralsService::ReferralInitializedCallback(
+            base::BindLambdaForTesting([&](const std::string& download_id) {
+              init_callback_called_ = true;
+              received_download_id_ = download_id;
+            })));
+  }
+
+  void TearDown() override {
+    referrals_service_.reset();
+    BraveReferralsService::SetPromoFilePathForTesting(base::FilePath());
+    BraveReferralsService::SetReferralInitializedCallbackForTesting(nullptr);
+  }
+
+  class TestBraveReferralsServiceDelegate
+      : public BraveReferralsService::Delegate {
+   public:
+    TestBraveReferralsServiceDelegate() = default;
+    ~TestBraveReferralsServiceDelegate() override = default;
+
+    void OnInitialized() override {}
+
+    base::FilePath GetUserDataDirectory() override { return user_data_dir_; }
+
+    network::mojom::URLLoaderFactory* GetURLLoaderFactory() override {
+      return loader_factory;
+    }
+
+#if !BUILDFLAG(IS_ANDROID)
+    base::OnceCallback<base::Time()> GetFirstRunSentinelCreationTimeCallback()
+        override {
+      return base::BindLambdaForTesting([this]() { return first_run_time_; });
+    }
+#endif
+
+    base::FilePath user_data_dir_;
+    base::Time first_run_time_;
+    raw_ptr<network::mojom::URLLoaderFactory> loader_factory = nullptr;
+  };
+
+  void CreateReferralsService() {
+    auto test_delegate = std::make_unique<TestBraveReferralsServiceDelegate>();
+    test_delegate->user_data_dir_ = temp_dir_.GetPath();
+    test_delegate->loader_factory = shared_url_loader_factory_.get();
+    test_delegate->first_run_time_ = first_run_time_;
+
+    referrals_service_ = std::make_unique<BraveReferralsService>(
+        &pref_service_, "test", "test-api-key");
+    referrals_service_->set_delegate(std::move(test_delegate));
+  }
+
+ protected:
+  content::BrowserTaskEnvironment task_environment_;
+  base::Time first_run_time_ = base::Time::Now();
+  network::TestURLLoaderFactory url_loader_factory_;
+  TestingPrefServiceSimple pref_service_;
+  scoped_refptr<network::SharedURLLoaderFactory> shared_url_loader_factory_;
+  base::ScopedTempDir temp_dir_;
+  std::unique_ptr<BraveReferralsService> referrals_service_;
+  size_t request_count_ = 0;
+  bool init_callback_called_ = false;
+  std::string received_download_id_;
+};
+
+TEST_F(BraveReferralsServiceTest, InitAndActivity) {
+  CreateReferralsService();
+  referrals_service_->Start();
+
+  task_environment_.FastForwardBy(base::Minutes(1));
```

Comment:
Q) Is this for making sure `OnReadPromoCodeComplete` is called?

Reply:
yes